boards/arm/nucleo_f746zg/doc/nucleof746zg.rst

```diff
@@ -0,0 +1,189 @@
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port                         |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| ETHERNET  | on-chip    | ethernet                            |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+``boards/arm/nucleo_f746zg/nucleo_f746zg_defconfig``
+
+For mode details please refer to `STM32 Nucleo-144 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+TBC
+
+
+System Clock
+------------
+
+Nucleo F746ZG System Clock could be driven by an internal or external oscillator,
+as well as the main PLL clock. By default, the System clock is driven by the PLL clock at 84MHz,
+driven by an 8MHz high-speed external clock.
+
+Serial Port
+-----------
+
+Nucleo F746ZG board has 4 UARTs and 4 USARTs. The Zephyr console output is assigned to UART2.
+Default settings are 115200 8N1.
+
+
+Programming and Debugging
+*************************
+
+Applications for the ``nucleo_f746zg`` board configuration can be built and
+flashed in the usual way (see `build_an_application` and
+`application_run` for more details).
+
+Flashing
+========
+
+Nucleo F746ZG board includes an ST-LINK/V2-1 embedded debug tool interface.
+
+Flashing an application to Nucleo F746ZG
+----------------------------------------
+
+Here is an example for the `hello_world` application.
+
+Run a serial host program to connect with your Nucleo board.
+
+.. code-block:: console
+
+   $ minicom -b 115200 -D /dev/ttyACM0
+
+Build and flash the application:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: nucleo_f746zg
+   :goals: build flash
+
+You should see the following message on the console:
+
+.. code-block:: console
+
+   $ Hello World! nucleo_f746zg
+
+Debugging
+=========
+
+You can debug an application in the usual way.  Here is an example for the
+`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: nucleo_f746zg
+   :maybe-skip-config:
+   :goals: debug
+
+.. _Nucleo F746ZG website:
+   https://www.st.com/en/evaluation-tools/nucleo-f746zg.html
+
+.. _STM32 Nucleo-144 board User Manual:
+   http://www.st.com/resource/en/user_manual/dm00105823.pdf
+
+.. _STM32F746ZG on www.st.com:
+   https://www.st.com/content/st_com/en/products/microcontrollers/stm32-32-bit-arm-cortex-mcus/stm32-high-performance-mcus/stm32f7-series/stm32f7x6/stm32f746zg.html
```

Comment:
this is a case where a long line length is OK.  Otherwise, please keep line length < 80